docs/rules/consistent-function-scoping.md

```diff
@@ -0,0 +1,44 @@
+# Prefer consistent function scoping
+
+A function definition should be placed as close to the top-level scope as possible without breaking its captured values.
+
```

Comment:
Can you mention that this makes the code more readable, and it can also have a performance benefit as JS engines can better optimize.

Reply:
If you happen to have a source for the JS optimization part I can also add a link. I hadn't heard that before.

Reply:
Sure. There's the obvious overhead of recreating the function for each call: https://stackoverflow.com/a/81329/64949 There's also a function size limit where the JIT optimizer bails out: https://ponyfoo.com/articles/javascript-performance-pitfalls-v8#optimization-limit